src/traccuracy/metrics/_track_overlap.py

```diff
@@ -0,0 +1,108 @@
+
+        - Target Effectiveness
+        - Track Purity
+
+        Args:
+            matched_data (Matched): Matched object for set of GT and Pred data
+            include_division_edges (bool, optional): If True, include edges at division.
+        """
+        self.include_division_edges = include_division_edges
+        super().__init__(matched_data)
+
+    def compute(self):
+        # requires tracklets that also have the splitting and merging edges
+        # edgess are a list of edges, grouped by the tracks
+        gt_tracklets = self.data.gt_graph.get_tracklets(
+            include_division_edges=self.include_division_edges
+        )
+        pred_tracklets = self.data.pred_graph.get_tracklets(
+            include_division_edges=self.include_division_edges
+        )
+
+        gt_pred_mapping = self.data.mapping
+        pred_gt_mapping = [
+            (pred_node, gt_node) for gt_node, pred_node in gt_pred_mapping
+        ]
+
+        # calculate track purity and target effectiveness
+        track_purity = _calc_overlap_score(
+            pred_tracklets, gt_tracklets, pred_gt_mapping
+        )
+        target_effectiveness = _calc_overlap_score(
+            gt_tracklets, pred_tracklets, gt_pred_mapping
+        )
+        return {
+            "track_purity": track_purity,
+            "target_effectiveness": target_effectiveness,
+        }
+
+
+def _calc_overlap_score(
+    reference_tracklets: List[TrackingGraph],
+    overlap_tracklets: List[TrackingGraph],
+    mapping: List[Tuple[Any, Any]],
+):
+    """Calculate weighted sum of the length of the longest overlap tracklet
+    for each reference tracklet.
+
+    Args:
+       reference_tracklets (List[TrackingGraph]): The reference tracklets
+       overlap_tracklets (List[TrackingGraph]): The tracklets that overlap
+       mapping (List[Tuple[Any, Any]]): Mapping between the reference tracklet nodes
+       and the overlap tracklet nodes
+
+    """
+    correct_count = 0
+    total_count = 0
+    # iterate over the reference tracklets
+
+    def map_node(node):
+        return [n_to for (n_from, n_to) in mapping if n_from == node]
+
+    for reference_tracklet in reference_tracklets:
+        # find the overlap tracklet with the largest overlap
+        reference_tracklet_edges_mapped = []
+        for node1, node2 in reference_tracklet.edges():
+            mapped_nodes1 = map_node(node1)
+            mapped_nodes2 = map_node(node2)
+            if mapped_nodes1 and mapped_nodes2:
+                for n1, n2 in product(mapped_nodes1, mapped_nodes2):
+                    reference_tracklet_edges_mapped.append((n1, n2))
+        overlaps = [
+            len(set(reference_tracklet_edges_mapped) & set(overlap_tracklet.edges()))
+            for overlap_tracklet in overlap_tracklets
+        ]
+        max_overlap = max(overlaps)
+        correct_count += max_overlap
+        total_count += len(reference_tracklet_edges_mapped)
```

Comment:
Is there a reason the total is computed using only the mapped edges instead of `len(reference_tracklet)`? Based on my understanding of the metric, we want to divide by the total length of the reference tracklet, which is different from the mapped edges if there is a false negative node (the `if` on line 97 is false).

Reply:
It would be good to test more cases like this to ensure there aren't any other edge cases we are missing - false negative nodes, false positive nodes, etc.

Reply:
Thanks for your comment! Yeah right, I believe it was not correct that I mapped the reference tracklet. I reorganized the code so that it maps the overlapping tracklet. I also added test with false negative nodes, false positive nodes and overlapping detections. I also checked the values of TE and TP swaps with swapping gt and pred tracks.